bootstrap/kubeadm/api/v1beta1/kubeadmconfig_webhook.go

```diff
@@ -43,9 +43,24 @@ func (c *KubeadmConfig) SetupWebhookWithManager(mgr ctrl.Manager) error {
 		Complete()
 }
 
+// +kubebuilder:webhook:verbs=create;update,path=/mutate-bootstrap-cluster-x-k8s-io-v1beta1-kubeadmconfig,mutating=true,failurePolicy=fail,matchPolicy=Equivalent,groups=bootstrap.cluster.x-k8s.io,resources=kubeadmconfigs,versions=v1beta1,name=default.kubeadmconfig.bootstrap.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
 // +kubebuilder:webhook:verbs=create;update,path=/validate-bootstrap-cluster-x-k8s-io-v1beta1-kubeadmconfig,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=bootstrap.cluster.x-k8s.io,resources=kubeadmconfigs,versions=v1beta1,name=validation.kubeadmconfig.bootstrap.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
 
-var _ webhook.Validator = &KubeadmConfig{}
+var (
+	_ webhook.Defaulter = &KubeadmConfig{}
+	_ webhook.Validator = &KubeadmConfig{}
+)
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type.
+func (c *KubeadmConfig) Default() {
+	defaultKubeadmConfigSpec(&c.Spec)
+}
+
+func defaultKubeadmConfigSpec(s *KubeadmConfigSpec) {
+	if s.Format == "" {
+		s.Format = CloudConfig
+	}
+}
```

Comment:
Could we default this field in the openapi schema spec rather than creating a webhook?

Reply:
Done, I think. Please have a look.